_source/_posts/2021-07-14-flask-scim-server.md

```diff
@@ -0,0 +1,519 @@
+
+#### Option 2: Enable SCIM Provisioning for Existing AIW App
+
+> Feel free to skip over this section to **Test the SCIM Server** if you set your SCIM integration up above.
+
+1. In your Okta dashboard, go to **Applications** -> **Applications**, then click the **Create App Integration** button. For this setup we will select **SWA - Secure Web Authentication**. Click **Next**.
+2. You can put whatever you'd like for the **App Name** and **App Login Page URL**, as we will just be loking at the SCIM functionality and not the SWA aspect of this app. Click **Finish**.
+3. In the **General** tab of the app, click **Edit** and toggle Provisioning from **None** to **SCIM**. Click **Save**.
+4. Your app should now have a **Provisioning** tab. Tab over to it and fill out the integration settings like the below image. Make the Authorization header **123456789**. You can change this later in the SCIM flask app.
+
+    {% img blog/flask-scim-server/scim_4.png alt:"AIW SCIM Settings Image" width:"800" %}{: .center-image }
+
+5. Click **Test Connector Configuration** and you should see the following success confirmation:
+
+    {% img blog/flask-scim-server/scim_5.png alt:"Test Connector Configuration UI Image" width:"800" %}{: .center-image }
+
+6. At which point, you can now click **Save**.
+
+    > You can navigate to `http://localhost:4040` to see the request from Okta on this request, as well as the response from the SCIM server.
+
+    {% img blog/flask-scim-server/scim_6.png alt:"Test Connector Configuration Call Image" width:"800" %}{: .center-image }
+
+
+7. Now your **Provisoning** tab will look a bit different. Click **Edit** next to **Provisioning to App** and check off:
+    - Create Users
+    - Update User Attributes
+    - Deactivate Users
+    
+And **Save**.
+
+{% img blog/flask-scim-server/scim_7.png alt:"AIW Provisioning Settings Image" width:"800" %}{: .center-image }
+
+You should now be set on the Okta side to start testing the SCIM server.
+
+{% img blog/flask-scim-server/frodo.webp alt:"AIW Provisioning Settings Image" width:"800" %}{: .center-image }
+
+### What did we just do?
+
+Before we go ahead and test our SCIM server, let's talk a bit about what we did in the setup steps above. Essentially we setup the following 3 components:
+
+{% img blog/flask-scim-server/flow_1.png alt:"SCIM FLow IMage" width:"1000" %}{: .center-image }
+
+In this setup, Okta is the source of truth for user data. The SCIM server opens up four endpoints locally which Okta is able to access thanks to the routing capabilities of ngrok. These endpoints are:
+
+- `https://localhost:5000/scim/v2/Users`
+- `https://localhost:5000/scim/v2/Users/{user_id}`
+- `https://localhost:5000/scim/v2/Groups`
+- `https://localhost:5000/scim/v2/Groups/{group_id}`
+
+
+These endpoints trigger CRUD operations based off the requests from Okta. Some examples of these calls:
+
+#### CREATE
+
+Okta Examples:
+
+- Assigning users to Okta app (`POST` to `/scim/v2/Users`)
```

Comment:
I suggest spelling this out more explicitly. For example, instead of the text above, use this:

> When a user is assigned to the app, Okta will make `POST` to `/scim/v2/Users`